### DISCUSSION TITLE
Tracking ticket for relevant events/workshops/hackathons/conferences/etc

### DISCUSSION THREAD

**Discussion by sanjayankur31** [OCNS/SoftwareWG#11] [General]
Please use this thread to share any events that you think would be relevant to the members of the WG.

CC: @OCNS/software-wg

**Comment by appukuttan-shailesh** [comment]
Let me kick off this thread with the first announcement:

**EBRAINS Infrastructure Training on Model Validation**
_4–7 May 2021 | Virtual Event_

For further details and registrations, please visit:
https://www.humanbrainproject.eu/en/education/training-on-model-validation/

This training will guide modellers and experimentalists in model validation (comparing simulation results to experimental data) using the EBRAINS Model Validation Framework.

Participants will learn how to:
- develop model-agnostic validation tests
- adapt existing models so they can be more easily validated
- use the EBRAINS model and test catalogue
- register, search, view and compare the results of validation tests

The training takes place over four days, with two days of presentations and hands-on demos, followed by two days of participants working on their own validation projects with the assistance of the tutors.

**Comment by malinINCF** [comment]
From @pgleeson (by email to comp-neuro et al)
### 
**3 NeuroML hackathon sessions at HARMONY 2021**

We will be running 3 online NeuroML hackathon sessions during the upcoming HARMONY 2021 meeting on 23-25th March. The general theme of the sessions will be: Learn to build, visualise, analyse and simulate your models using NeuroML.

**Why take part?**
These hackathons will give members of the modelling community the chance to:

- Get high level introductions to the NeuroML language and toolchain
- Meet the NeuroML core development team and editors
- Find out the latest information on which simulators/applications support NeuroML
- Open, discuss and work on issues related to converting your model to NeuroML, or supporting NeuroML in your simulator
- Learn how to share your models with the community 

**Times and dates**
All sessions will be online and take place over 3 hours (9am-noon Pacific; 12-3pm EST time; 4-7pm UK/UTC; 5-8pm CET, 9:30pm-12:30am IST; note non-standard US/EU time differences that week). The broad focus of each of the sessions (dependent on interests of attendees) is:

- Tues 23rd March: Introduction to NeuroML, general questions about usage
- Wed 24th March: Detailed cell/conductance based models (e.g. converting channels to NeuroML)
- Thus 25th March: Abstract/point neuron networks including PyNN interactions

**Registration**
To take part in the hackathon, please register at [http://co.mbine.org/events/HARMONY_2021](http://co.mbine.org/events/HARMONY_2021) for the HARMONY meeting (registration is free). You will get sent details to access the agenda on [https://harmony2021.sched.com](https://harmony2021.sched.com), which will have links to the Zoom sessions for each of the days. 

**Open an issue beforehand!**
While it will be possible to raise and discuss new issues at the hackathons, it will be easier to manage and plan work/discussions if you open an issue with a description of the problem you are trying to address at: [https://github.com/NeuroML/NeuroML2/issues](https://github.com/NeuroML/NeuroML2/issues). 

**Slack**
To aid communication with the community during (and after) the meeting, we have a Slack channel for NeuroML related discussions. Please contact @pgleeson for an invite

We look forward to working with the community to drive further uptake of NeuroML compliant models and tools!

**Comment by sanjayankur31** [comment]
**[On the evaluation of research software: the CDUR procedure - SORSE](https://sorse.github.io/programme/talks/event-039/)**

March 16, 2021, 14:00 – 14:30 UTC   2021-03-16T14:00:00+00:00 2021-03-16T14:30:00+00:00

The goal of the proposed talk is to present the publication

Gomez-Diaz T. and Recio T., On the evaluation of research software: the CDUR procedure, [version 2 ; peer review: 2 approved] 26 Nov 2019. F1000Research 2019, 8:1353, https://f1000research.com/articles/8-1353

**Background**: Evaluation of the quality of research software is a challenging and relevant issue, still not sufficiently addressed by the scientific community.

**Methods**: Our contribution begins by defining, precisely but widely enough, the notions of research software and of its authors followed by a study of the evaluation issues, as the basis for the proposition of a sound assessment protocol: the CDUR procedure.

**Results**: CDUR comprises four steps introduced as follows: Citation, to deal with correct RS identification, Dissemination, to measure good dissemination practices, Use, devoted to the evaluation of usability aspects, and Research, to assess the impact of the scientific work.

**Conclusions**: Some conclusions and recommendations are finally included. The evaluation of research is the keystone to boost the evolution of the Open Science policies and practices. It is as well our belief that research software evaluation is a fundamental step to induce better research software practices and, thus, a step towards more efficient science.

**Comment by sanjayankur31** [comment]
Folks, the [Open Neuroscience](https://open-neuroscience.com/) folks are organising talks related to Open Source tools on the [World Wide Neuro](https://www.world-wide.org/) platform. Please see their tweet here on how to contact them about a session:

https://twitter.com/openneurosci/status/1370060159985324037

**Comment by sanjayankur31** [comment]
[Brainhack: Developing a culture of open, inclusive, community-driven neuroscience: Neuron](https://www.cell.com/neuron/fulltext/S0896-6273(21)00231-2)

**Comment by sanjayankur31** [comment]
Fedora is slowly moving to Zuul for CI. It's developed by the "Open Infrastructure project": 

https://www.openstack.org/news/view/426/kata-containers-and-zuul-are-first-pilot-projects-confirmed-as-toplevel-open-infrastructure-projects-by-the-openstack-foundation-board

I haven't checked to see if there are hosted instances available yet, but here is the URL in the meantime:

https://zuul-ci.org/

https://zuul.opendev.org/t/openstack/status

Here's the Fedora wiki page on the subject: https://fedoraproject.org/wiki/Zuul-based-ci

**Comment by sanjayankur31** [comment]
@malinINCF 's post on Brainhack:

[10+ years of Brainhack: an open, inclusive culture for neuro tool developers at all levels – Malins blogg.](https://malinsandstrom.blog/2021/05/12/10-years-of-brainhack-an-open-inclusive-culture-for-neuro-tool-developers-at-all-levels/)

**Reply by malinINCF** [reply]
The preprint itself is here, [https://psyarxiv.com/rytjq/](https://psyarxiv.com/rytjq/ )

**Comment by sanjayankur31** [comment]
We are happy to announce the upcoming Organization for Human Brain Mapping (OHBM) Brainhack 2021 happening June 16-18 around the clock around the world! Join the party by registering [here](https://www.humanbrainmapping.org/i4a/ams/meetings/index.cfm?controller=meetings&action=startRegistration&conferenceID=136&reginit=1&pageID=4073) ($25 fee, or optional fee waiver; check out some project proposals [here](https://ohbm.github.io/hackathon2021/hacktrack/)).

We are also looking for volunteers to help spread the word and facilitate Brainhack activities. We’d like to recruit around 100 volunteers—so any contribution is more than welcome, no prerequisites needed! Volunteer by filling out [this form](https://tinyurl.com/BH2021volunteer) (more information about volunteer roles [here](https://docs.google.com/document/d/10TT55AXcC8Z1CsXMgTKqM5dIl6xpmT4ZrhMDP2j2LG8/edit)).

About Brainhack:
[Brainhacks](https://brainhack.org/) (including [OHBM Brainhack](https://ohbm.github.io/hackathon2021/)) are events promoting interdisciplinary neuroscience research; they bring researchers together from a variety of subfields (e.g., neurobiology, psychology, computer science, etc.) and research backgrounds (e.g., medical student, industry professional, RA, etc.) to collaborate on projects and ideas and teach open science-oriented technical skills. The focus on intensive community- and skill-building opens doors to a number of professional opportunities (learn more about these benefits [here](https://www.sciencedirect.com/science/article/pii/S0896627321002312)). Brainhacks are not your typical hackathon—they are actively inclusive rather than competitive. To that end, even if you have never attended a hackathon before or written a line of code, you are welcome! No need to work with neuroimaging methods or be part of OHBM to attend. All you need is to come along and share your ideas, expertise, contributions with whatever pace and form you can make it happen while having a chance to collaborate and meet with many other researchers all around the world!  

We particularly welcome the contributions of groups traditionally underrepresented in neuroscience, including women, BIPOC, LGBTQ+, individuals with disabilities, and more. Brainhacks are safe spaces, with a policy of zero tolerance for harassment. OHBM Brainhack 2021 is intended for everyone to feel safe, learn from each other and share knowledge. Our event would be enriched by your involvement and contributions. To support our mission, please share this widely—we’d especially love your help translating and sharing this message in your local language to reach as many people around the world as we can! Events will take place across two main time slots to support as many time zones as possible.

About us:
We are the Open Science Special Interest Group (OSSIG) within the Organization for Human Brain Mapping (OHBM) society, which hosts a yearly conference on mapping the brain with neuroimaging. The OSSIG is dedicated to promoting Open Science in this community and neuroscience more broadly.

Please contact ohbmopenscience AT gmail DOT com if you have any questions.

**Comment by sanjayankur31** [comment]
https://neurotechai.eu/educational/ on June 8, 4PM CEST:

HOW CAN WE SIMULATE SPIKING NEURAL NETWORKS?

Schedule Day 8:

- Dr. Marcel Stimberg (Brian simulator) (@mstimberg )
- Dr. Hans Ekkehard Plesser (NEST simulator)
- Prof. Thomas Nowotny (University of Sussex) (@tnowotny )
- Chair: Prof. Michael Schmuker (University of Hertfordshire, UK)

**Comment by sanjayankur31** [comment]
Was looking at the Software Carpentry docs and ran into this:

https://carpentries.github.io/instructor-training/

I think it'll be great to do this, I'll see when I can find the time :)

**Comment by sanjayankur31** [comment]
The next WG dev session is on June 21 at 1600 UTC on SciUnit. Please see the details here. I've also sent the announcement to the usual mailing lists

https://ocns.github.io/SoftwareWG/2021/06/04/dev-session-rick-gerkin-sciunit.html

**Reply by sanjayankur31** [reply]
Of course, we hope to see you all the dev session :)

**Reply by sanjayankur31** [reply]
Zoom link for session, starting in less than an hour: https://ucl.zoom.us/j/91246735902?pwd=WVdVK3I5SGdlaDFZdllDQ0w2djVBdz09

**Reply by sanjayankur31** [reply]
I've started the zoom meeting now folks, please join at your convenience.

**Comment by sanjayankur31** [comment]
This looks useful:

- [torbiak/git-autofixup: create fixup commits for topic branches](https://github.com/torbiak/git-autofixup)
- [Effortlessly correct your Git commits with git-autofixup](https://symflower.com/en/company/blog/2021/git-autofixup/)

**Comment by sanjayankur31** [comment]
Ran into this recently:

[Responses to 10 common criticisms of anti-racism action in STEMM](https://journals.plos.org/ploscompbiol/article?id=10.1371%2Fjournal.pcbi.1009141)

I've been wondering how we can increase the diversity of the WG membership. We can discuss it at a meeting when everyone has returned.

**Comment by sanjayankur31** [comment]
[Announcing State of Software Security v11: Open Source Edition| Veracode...](https://www.veracode.com/blog/research/announcing-state-software-security-v11-open-source-edition)

**Comment by sanjayankur31** [comment]
Just a heads up folks: I've reserved a Gitter room for us here: https://gitter.im/OCNS/SoftwareWG. It's a public room so please feel free to drop by and chat there---troubleshooting etc. and any software related discussion is all welcome. If you're already using Matrix/Element, you can also use that to access the room using the web client, or one of their desktop/mobile clients: https://matrix.to/#/%23OCNS_SoftwareWG:gitter.im?utm_source=gitter

This is just so folks have a place to chat if ever necessary (and so that we don't ever think of creating a Slack workspace (Slack is not FOSS and it's channels are not public so we shouldn't use it, at least not for this WG :)))

**Comment by sanjayankur31** [comment]
Not much new here, but worth skimming:

[Ten simple rules on writing clean and reliable open-source scientific software](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009481)

**Comment by sanjayankur31** [comment]
Ran into Toolbx recently, which is meant to make container based development easier. Worth a look:

- [toolbx](https://containertoolbx.org/)
- [Toolbx, a developers new best friend! - Fedora Magazine](https://fedoramagazine.org/toolbx-a-developers-new-best-friend/)

"Toolbx is a tool for Linux operating systems, which allows the use of containerized command line environments. It is built on top of Podman and other standard container technologies from OCI."

**Comment by sanjayankur31** [comment]
Preparations for FOSDEM are on. "FOSDEM is a free event for software developers to meet, share ideas and collaborate."

[FOSDEM 2022 - Home](https://fosdem.org/2022/)

It's free, and online on Feb 5-6.

There's also the Open research tools and tech devroom, but it's not been accepted yet:

- [Join us at FOSDEM 2022 | Open Research Tools & Technologies](https://research-fosdem.github.io/)

Mailing list for announcements: https://lists.fosdem.org/listinfo/fosdem

**Reply by sanjayankur31** [reply]
The open research tools and tech devroom is now open for submissions:

https://research-fosdem.github.io/

CALL FOR PROPOSAL:
Join us at FOSDEM 2022!
Saturday 5th of February 2022 - ONLINE
(precise time to be announced)

We invite developers and users of open tools and technologies used in a research and investigation context to contribute to the 2022 edition of FOSDEM, the largest open-source conference in Europe. We understand research as the general process of knowledge production and inquiry. This includes scientific research, investigative journalism, data journalism, as well as research and investigations undertaken by NGOs, civil society, community and activist groups, etc.

**Comment by sanjayankur31** [comment]
Just got an e-mail about this:

[INCF endorses Five Recommendations for FAIR Software as a Best Practice | INCF](https://www.incf.org/blog/incf-endorses-five-recommendations-fair-software-best-practice)

The tool is here:

https://fair-software.eu/

**Comment by sanjayankur31** [comment]
I don't know if I posted this before: a free course from the Linux Foundation:

[Inclusive Open Source Community Orientation (LFC102) - Linux Foundation - Training](https://training.linuxfoundation.org/training/inclusive-open-source-community-orientation-lfc102/)

**Comment by sanjayankur31** [comment]
[Ten simple rules to ruin a collaborative environment](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009957)

**Reply by appukuttan-shailesh** [reply]
Seems interesting!

**Comment by sanjayankur31** [comment]
[Supporting academic equity in physics through citation diversity | Communications Physics](https://www.nature.com/articles/s42005-022-00999-9)

**Comment by sanjayankur31** [comment]
[Ten simple rules for successfully supporting first-generation/low-income (FLI) students in STEM | PLOS Computational Biology](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010499)

[Ten simple rules for getting and giving credit for data | PLOS Computational Biology](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010476)

**Comment by sanjayankur31** [comment]
[Ten simple rules for funding scientific open source software | PLOS Computational Biology](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010627): seems very relevant to quite a few of us.

**Reply by mstimberg** [reply]
> [Ten simple rules for funding scientific open source software | PLOS Computational Biology](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010627) : seems very relevant to quite a few of us.

Just a quick comment, since I got misled by the title: this is directed at funders and not developers, i.e. it is about "how to fund software" not "how to get funded" :blush:

**Comment by sanjayankur31** [comment]
This looks interesting:

https://github.com/FAIR-CA-indicators/CA-RDA-Indicators/issues

More info here:

https://groups.google.com/g/combine-discuss/c/_LAvGSNtmtQ

**Reply by sanjayankur31** [reply]
^ [Invitation to hackathons on FAIR model indicators (Feb 17-Mar 3)](https://groups.google.com/g/combine-discuss/c/_LAvGSNtmtQ)

**Comment by sanjayankur31** [comment]
[How to Sustain a Scientific Open-Source Software Ecosystem: Learning from the Astropy Project](https://arxiv.org/abs/2402.15081)

**Comment by mstimberg** [comment]
Didn't look into it further, but might be interesting (for those not on summer vacation during that time): 
> Dear all,
> The Common Fund SPARC program, with funding from the National Institutes of Health (NIH), has a three-day in-the-cloud Codeathon that may be of interest for the comp-neuro community.
> The fourth annual virtual SPARC Codeathon will take place online August 10-12. This global prize competition is currently recruiting participants to make [SPARC data and resources](https://sparc.science/) more Findable, Interoperable, Accessible, and Reusable (FAIR). Any interested participant can join projects spanning data visualization, simulation tool development, python workflow creation, and more to compete for $50,000 in prizes.
> Follow this link to learn more and sign up today! https://sparc.science/news-and-events/events/2024-sparc-fair-codeathon

**Reply by sanjayankur31** [reply]
That does indeed look very interesting.

**Comment by sanjayankur31** [comment]
[Software in science is ubiquitous yet overlooked | Nature Computational Science](https://www.nature.com/articles/s43588-024-00651-2)
